aws/logs_monitoring/enhanced_lambda_metrics.py

```diff
@@ -254,6 +259,89 @@ def parse_lambda_tags_from_arn(arn):
     ]
 
 
+MEMORY_ALLOCATED_RECORD_KEY = "memorySizeMB"
+INIT_DURATION_RECORD_KEY = "initDurationMs"
+BILLED_DURATION_RECORD_KEY = "billedDurationMs"
+RUNTIME_METRICS_BY_RECORD_KEY = {
+    # Except INIT_DURATION_RECORD_KEY which is handled separately
+    "durationMs": DURATION_METRIC_NAME,
+    BILLED_DURATION_RECORD_KEY: BILLED_DURATION_METRIC_NAME,
+    "maxMemoryUsedMB": MAX_MEMORY_USED_METRIC_NAME,
+}
+
+
+def parse_metrics_from_json_report_log(log_message):
+    if not log_message.startswith("{"):
+        return []
```

Comment:
nit: no need for this here, we can return at the try/except of parsing json fails